MapboxCoreNavigationTests/RouteControllerTests.swift

```diff
@@ -81,4 +83,36 @@ class RouteControllerTests: XCTestCase {
         XCTAssertEqual(directionToStart, navigation.location!.course, "The course should be the raw course and not an interpolated course")
         XCTAssertFalse(facingTowardsStartLocation.shouldSnap(toRouteWith: facingTowardsStartLocation.interpolatedCourse(along: navigation.routeProgress.currentLegProgress.nearbyCoordinates)!, distanceToFirstCoordinateOnLeg: facingTowardsStartLocation.distance(from: firstLocation)), "Should not snap")
     }
+    
+    func testLocationShouldUseHeading() {
+        let navigation = setup.routeController
+        let firstLocation = setup.firstLocation
+        navigation.locationManager(navigation.locationManager, didUpdateLocations: [firstLocation])
+        
+        XCTAssertEqual(navigation.location!.course, firstLocation.course, "Course should be using course")
+        
+        let invalidCourseLocation = CLLocation(coordinate: firstLocation.coordinate, altitude: firstLocation.altitude, horizontalAccuracy: firstLocation.horizontalAccuracy, verticalAccuracy: firstLocation.verticalAccuracy, course: -1, speed: firstLocation.speed, timestamp: firstLocation.timestamp)
+        
+        let heading = CustomHeading()
```

Comment:
There is no initializer for CLHeading so I created my own. Unsure about this move.

Reply:
@bsudekum I'm reviewing this PR. Btw, what are you unsure of?

Reply:
@vincethecoder whether we want to use this custom class I created or if we should somehow create our own initializer.

Reply:
@bsudekum rather than a subclass, I would create an `extension` of `CLHeading` like so:

```
extension CLHeading {
    
    var trueHeading: CLLocationDirection {
        set {
            mbTrueHeading = newValue
        }
        get {
            return mbTrueHeading
        }
    }

    private var mbTrueHeading: CLLocationDirection {
        set {
            trueHeading = newValue
        }
        get {
            return trueHeading
        }
    }
}
```

Reply:
Unfortunately, CLHeading’s properties are not mutable so I don't think we can use an extension for that. Since this is in the test target, we could expose CLHeading’s private initializer in ObjC.

```objc
@interface CLHeading (Private)
- (id)initWithHeading:(double)arg1 accuracy:(double)arg2;
@end
```

Reply:
@frederoni 
> Unfortunately, CLHeading’s properties are immutable

Yes, indeed. My suggestion was, 

> Since this is in the test target...

We could get away with either one of these suggested approaches:

```
fileprivate var mbTrueHeading: CLLocationDirection = 50

extension CLHeading {
    
    // Approach 1: Initializer
    convenience init(_ heading: CLLocationDirection) {
        self.init()
        mbTrueHeading = heading
    }
    
    // Approach 2: A getter and setter
    var trueHeading: CLLocationDirection {
        get {
            return mbTrueHeading
        }
        set {
            mbTrueHeading = newValue
        }
    }
}
```

/cc @bsudekum

Reply:
@bsudekum ☝️ either a `// Approach 1: Initializer` or `// Approach 2: A getter and setter` above should suffice.